gk/main.c

```diff
@@ -180,10 +186,18 @@ extract_packet_info(struct rte_mbuf *pkt, struct ipacket *packet)
 	return ret;
 }
 
+static inline uint64_t
+calc_request_expire_at(uint64_t now)
```

Comment:
This function only calculates the initial expiration for request entries, so a better name is `calc_initial_request_expiration()`.

Reply:
I just noticed that when we set `above_priority = priority + 4`, the initial and the following priorities aligned, so we can have a version of `calc_request_expire_at()` for the initial and following states of request entries. What's needed is for `calc_request_expire_at()` to receive the priority assigned to the entry.

Reply:
Sorry, I don't know what you mean by "the initial and the following priorities aligned." Are you saying we can write one `calc_request_expire_at()` that computes the expiration according to the calculation based on `above_priority = priority + 4` that works for all priorities, even the initial priority?

Reply:
Yes.